ta4j/src/main/java/org/ta4j/core/indicators/pivotpoints/StandardReversalIndicator.java

```diff
@@ -0,0 +1,137 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package org.ta4j.core.indicators.pivotpoints;
+
+import org.ta4j.core.Decimal;
+import org.ta4j.core.indicators.RecursiveCachedIndicator;
+
+import java.util.List;
+
+/**
+ * Pivot Reversal Indicator.
+ * <p>
+ * @author team172011(Simon-Justus Wimmer), 11.10.2017
+ * @see <a href="http://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:pivot_points">chart_school: pivotpoints</a>
+ */
+public class StandardReversalIndicator extends RecursiveCachedIndicator<Decimal> {
+
+    private PivotPointIndicator pivotPointIndicator;
+    private PivotLevel level;
+
+    public enum PivotLevel{
+        RESISTANCE_3, RESISTANCE_2, RESISTANCE_1, SUPPORT_1, SUPPORT_2, SUPPORT_3;
+    }
+
+    /**
+     * Constructor.
+     * <p>
+     * Calculates the (standard) reversal for the corresponding pivot level
+     * @param pivotPointIndicator the {@link PivotPointIndicator} for this reversal
+     * @param level the {@link PivotLevel} for this reversal
+     */
+    public StandardReversalIndicator(PivotPointIndicator pivotPointIndicator, PivotLevel level) {
+        super(pivotPointIndicator);
+        this.pivotPointIndicator = pivotPointIndicator;
+        this.level =level;
+    }
+
+    @Override
+    protected Decimal calculate(int index) {
+        List<Integer> ticksOfPreviousPeriod = pivotPointIndicator.getTicksOfPreviousPeriod(index);
+        if (ticksOfPreviousPeriod.isEmpty())
+            return Decimal.NaN;
+        switch (level){
+            case RESISTANCE_3:
+                return calculateR3(ticksOfPreviousPeriod, index);
+            case RESISTANCE_2:
+                return calculateR2(ticksOfPreviousPeriod, index);
+            case RESISTANCE_1:
+                return calculateR1(ticksOfPreviousPeriod, index);
+            case SUPPORT_1:
+                return  calculateS1(ticksOfPreviousPeriod, index);
+            case SUPPORT_2:
+                return calculateS2(ticksOfPreviousPeriod, index);
+            case SUPPORT_3:
+                return calculateS3(ticksOfPreviousPeriod, index);
+            default:
+                return Decimal.NaN;
+        }
+
+    }
+
+    private Decimal calculateR3(List<Integer> ticksOfPreviousPeriod, int index){
+        Decimal low = getTimeSeries().getTick(ticksOfPreviousPeriod.get(0)).getMinPrice();
+        Decimal high =  getTimeSeries().getTick(ticksOfPreviousPeriod.get(0)).getMaxPrice();
```

Comment:
no need to query again and again for tick, put Tick tick = getTimeSeries().getTick(ticksOfPreviousPeriod.get(0)); Look at https://github.com/ta4j/ta4j/pull/49.